### ISSUE TITLE
Can I change the training dataset to segmentation other things like portrait?

### ISSUE THREAD

**Issue by leijue222** [milesial/Pytorch-UNet#132]
First:
I put 1700 portrait images into './data/imgs/'
and  put 1700 target masks images into './data/masks/'

Then I rented a cloud GPU server(RTX 2080Ti) and use the default configuration to run:

> python train.py 

![1582211216](https://user-images.githubusercontent.com/30276789/74947410-c696e880-5435-11ea-8abc-97c931d3889e.png)

I successed and I recorded it on [Youtube](https://youtu.be/etVvPmNnPqI)
Please adjust the video quality to 1080P and watch the left half of the video.

But when I run:

> python predict.py -i car1.jpg image.jpg --viz --no-save --model ./checkpoints/CP_epoch5.pth

I get this result:
![8ea6d55c6815ab1d9dec920cd3203a8](https://user-images.githubusercontent.com/30276789/74943788-ec20f380-542f-11ea-9802-754f5bfcdb87.png)

I don't know where is the problem.
Could you please give me some helps?
Thanks!

**Comment by Branikas** [comment]
You need to train your model more, you dice coefficient is almost zero in 4 epochs out of 5. I would try something like 100 epochs or more and maybe a smaller learning rate. Your network hasn't learned and features and it predicts 0.

**Comment by leijue222** [comment]
@Branikas 
Thank you very much for your answer.
I have two questions:
1：These parameters: epochs, batch-size, learning-rate, scale, validation.
What value should I take?Could you give you parameters to me as reference.
Such as

> python train.py -e x -b x -l x -s x -v x

2：I find another problem, maybe about my dataset.
My images is 600*800 with .png format.
Where I run milesial's MODEL.pth.
I get this result:

> Given groups=1, weight of size 64 3 3 3, expected input[1, 4, 600, 600] to have 3 channels, but got 4 channels instead

I need to convert these png images from rgba to rgb probabely.

**Comment by leijue222** [comment]
> You need to train your model more, you dice coefficient is almost zero in 4 epochs out of 5. I would try something like 100 epochs or more and maybe a smaller learning rate. Your network hasn't learned and features and it predicts 0.

Thanks a million!
According to your proposal, set the epoch value to 100 and the scale value to 1.

> optional arguments:
   epochs         100  (default: 5)
  batch-size     1      (default: 1)
  learning-rate 0.1   (default: 0.1)
  scale             1      (default: 0.5)
  validation     15    (default: 15.0)
`python train.py -e 100 -b 1 -l 0.1 -s 1 -v 15`
It worked!

When loss (batch) = 0.55 around or >0.6, the result will be all black.
In 100 epochs, it have 5 epochs value in 0.2~0.3.
I have test the five models, the result is:
![soft_models](https://user-images.githubusercontent.com/30276789/75101557-3618fd80-5619-11ea-9d7e-3ef4e0cf8d5d.png)

And this is part of the training logs:
![training_log](https://user-images.githubusercontent.com/30276789/75101564-53e66280-5619-11ea-992c-e6e043472cb2.png)
We can see that at the end of 100 epochs, the loss value is still very volatile.
This shows that the convergence effect is not good.
In #95 . liaojiacai said he solved it by replace the loss function.
But did‘t say which loss function was use to replace the nn.BCEWithLogitsLoss().

Do you have a better idea for the situation I described?
For example：
 **1：Replace the loss function (I don't know use which loss function)
 2：Modify the values of epoch, batch size, lr, sacle, etc.**

**Comment by leijue222** [comment]
I have tried to modify the lr to 0.01,batch_size to 8. But it didn't have visible improvement.
I now want to try image enhancement methods to increase the number of training sets.
I hope that after doing so, the effect of training will be improved.

**Comment by leijue222** [comment]
18000 image and 18000 target masks (600*800)
> optional arguments:
> epochs 200 (default: 5)
> batch-size 1 (default: 1)
> learning-rate 0.1 (default: 0.1)
> scale 1 (default: 0.5)
> validation 15 (default: 15.0)
> `python train.py -e 200 -b 1 -l 0.1 -s 1 -v 15`
The result:
![Snipaste_2020-03-04_18-29-34](https://user-images.githubusercontent.com/30276789/75870817-86c4fd80-5e46-11ea-81eb-30ab3d775ca2.png)
It still doesn't look perfect.
Could anybody give some suggestions to improve the model?
@milesial  @Branikas

**Comment by chris3will** [comment]
我想问一下，标签是一定要用csv格式吗，可不可以用二值的图片作为标签呢

**Comment by leijue222** [comment]
可以呀，我训练用的是沈小勇Deep Auto Portrait Matting的数据集，这个数据集的标签就是二值的图片呀。

**Comment by chris3will** [comment]
好滴，我发现了还需要在train和predict中改一些代码，现在可以跑通了。蟹蟹呀

**Comment by yarkable** [comment]
> 
> 
> 我想问一下，标签是一定要用csv格式吗，可不可以用二值的图片作为标签呢

hhh，bro，your Chinese is so hardcore.

**Comment by bulingda** [comment]
How did you solve the problem that output_mask has no content?

**Comment by leijue222** [comment]
@bulingda Branikas has given the answer:  You need to train your model more, you dice coefficient is almost zero in 4 epochs out of 5. I would try something like 100 epochs or more and maybe a smaller learning rate. Your network hasn't learned and features and it predicts 0.

**Comment by bulingda** [comment]
Thank you very much！

**Comment by kobe233333** [comment]
> 可以呀，我训练用的是沈小勇Deep Auto Portrait Matting的数据集，这个数据集的标签就是二值的图片呀。

请问下mask的输入就是二值图片吗？那需要改代码中的channel数吗

**Comment by kobe233333** [comment]
> 好滴，我发现了还需要在train和predict中改一些代码，现在可以跑通了。蟹蟹呀

请问需要修改什么代码呢？是修改channel数吗

**Comment by JiashengHong** [comment]
How did you set LR policy when training so many epochs？
If using default LR policy ‘ReduceLROnPlateau’, I find LR will reduce quickly. And network can't learn anything.

**Comment by pcicales** [comment]
@JiashengHong which LR policy did you use? Experiencing a similar issue:
![Screenshot from 2020-05-26 18-27-21](https://user-images.githubusercontent.com/9851953/82959928-9c961a80-9f7e-11ea-89ff-a4de2580f1b6.png)

**Comment by JiashengHong** [comment]
@pcicales U-Net does not seem to require a high LR. I use a simple constant LR 1e-4 and train more epochs. It works in my task and my network can converge and overfit.

**Comment by pcicales** [comment]
@JiashengHong I guess it can depend on the data but yes it looks like the scheduler is not really necessary. I think I will still use it to control training on my data but mainly just to slow over fitting at later epochs. I also noticed that they used RMSprop which is odd; I changed it to the Adam optimizer and had significantly improved results.

**Comment by shilei2403** [comment]
> 好滴，我发现了还需要在train和predict中改一些代码，现在可以跑通了。蟹蟹呀
你好，我想问一下，你都修改了哪些地方？然后我的mask都是tif格式的，读取时候总是报错，你知道怎么读取TIF格式的图片么？

**Comment by shilei2403** [comment]
@chris3will
